std/experimental/allocator/common.d

```diff
@@ -743,3 +743,135 @@ version(unittest)
         }
     }
 }
+
+/+
+Can the representation be determined at compile time to consist of nothing but
+zero bits? Padding between a struct's fields is not considered.
++/
+private template isAllZeroBits(T, T value)
+{
+    import std.traits : isAggregateType, isDynamicArray, isStaticArray;
+    static if (isDynamicArray!(typeof(value)))
+        enum isAllZeroBits = value is null && value.length == 0;
+    else static if (is(typeof(value is null)))
+        enum isAllZeroBits = value is null;
+    else static if (is(typeof(value is 0)))
+        enum isAllZeroBits = value is 0;
+    else static if (isStaticArray!(typeof(value)))
+    {
+        static if (value.length == 0)
+            enum isAllZeroBits = true;
+        else
+        {
+            alias E = typeof(value[0]);
+            enum isAllZeroBits = mixin(()
+            {
+                enum N = typeof(value).length;
+                char[] buffer = new char[32 * value.length]; // Pre-size for value.length <= 999.
+                buffer.length = 0;
+                buffer ~= "isAllZeroBits!(E,value[0])";
+                import std.conv : to;
+                foreach (i; 1 .. N)
+                    buffer ~= " && isAllZeroBits!(E,value["~to!string(i)~"])";
+                return buffer;
+            }());
+        }
```

Comment:
Is there a more elegant way to check all the elements of a static array?

Reply:
And I think we can do even better here:

```d
enum isAllZeroBits = isAllZeroBits!(typeof(value[0]), value[0]);
```

Static arrays need to have the same type, don't they?

Reply:
That works only when the static array is the outermost type and not a field of another struct . That optimization could go in `isInitAllZeroBits` though.